apiv2/project/project.go

```diff
@@ -0,0 +1,540 @@
+package project
+
+import (
+	"context"
+	"errors"
+
+	modelv2 "github.com/mittwald/goharbor-client/apiv2/model"
+	uc "github.com/mittwald/goharbor-client/apiv2/user"
+
+	v2client "github.com/mittwald/goharbor-client/apiv2/internal/api/client"
+	"github.com/mittwald/goharbor-client/apiv2/internal/legacyapi/client"
+
+	"github.com/go-openapi/runtime"
+	"github.com/mittwald/goharbor-client/apiv2/internal/legacyapi/client/products"
+	model "github.com/mittwald/goharbor-client/apiv2/model/legacy"
+)
+
+const (
+	ProjectMetadataKeyEnableContentTrust   MetadataKey = "enable_content_trust"
+	ProjectMetadataKeyAutoScan             MetadataKey = "auto_scan"
+	ProjectMetadataKeySeverity             MetadataKey = "severity"
+	ProjectMetadataKeyReuseSysCVEWhitelist MetadataKey = "reuse_sys_cve_whitelist"
+	ProjectMetadataKeyPublic               MetadataKey = "public"
+	ProjectMetadataKeyPreventVul           MetadataKey = "prevent_vul"
+	ProjectMetadataKeyRetentionID          MetadataKey = "retention_id"
+)
+
+// RESTClient is a subclient for handling project related actions.
+type RESTClient struct {
+	// The legacy swagger client
+	LegacyClient *client.Harbor
+
+	V2Client *v2client.Harbor
+
+	// AuthInfo contains the auth information that is provided on API calls.
+	AuthInfo runtime.ClientAuthInfoWriter
+}
+
+func NewClient(legacyClient *client.Harbor, v2Client *v2client.Harbor, authInfo runtime.ClientAuthInfoWriter) *RESTClient {
+	return &RESTClient{
+		LegacyClient: legacyClient,
+		V2Client:     v2Client,
+		AuthInfo:     authInfo,
+	}
+}
+
+type Client interface {
+	NewProject(ctx context.Context, name string, countLimit int, storageLimit int)
+	DeleteProject(ctx context.Context, p *model.Project) error
+	GetProject(ctx context.Context, name string) (*model.Project, error)
+	ListProjects(ctx context.Context, nameFilter string) ([]*model.Project, error)
+	UpdateProject(ctx context.Context, p *model.Project, countLimit int, storageLimit int) error
+
+	AddProjectMember(ctx context.Context, p *model.Project, u *model.User, roleID int) error
+	ListProjectMembers(ctx context.Context, p *model.Project) ([]*model.ProjectMemberEntity, error)
+	UpdateProjectMemberRole(ctx context.Context, p *model.Project, u *model.User, roleID int) error
+	DeleteProjectMember(ctx context.Context, p *model.Project, u *model.User) error
+
+	AddProjectMetadata(ctx context.Context, p *model.Project, key MetadataKey, value string) error
+	ListProjectMetadata(ctx context.Context, p *model.Project) (*model.ProjectMetadata, error)
+	GetProjectMetadataValue(ctx context.Context, p *model.Project, key MetadataKey) (string, error)
+	UpdateProjectMetadata(ctx context.Context, p *model.Project, key MetadataKey, value string) error
+	DeleteProjectMetadataValue(ctx context.Context, p *model.Project, key MetadataKey) error
+
+	// v2 methods
+	GetProjectV2(ctx context.Context, projectID int64) (*modelv2.Project, error)
+
```

Comment:
I think we should not provide something like a `DoSomeThingV2` method but alter the old `DoSomeThing` method to use the v2 api of harbor. The purpose of this client is to abstract away Harbor api implementation details, so we should not confuse users with two ways of doing the same thing just because Harbor's api allow for it.